### DISCUSSION TITLE
Raylib on Re:Terminal

### DISCUSSION THREAD

**Discussion by cgn1234** [raysan5/raylib#2495] [General]
I’m trying to build raylib on the Re:Terminal (Raspberry PI CM4 based), but running into trouble. 

Documentation mentions that by default it builds without X Windows, but all of my builds have PLATFORM=PLATFORM_DRM. I tried modifying many files, but can’t seemed to figure out the right way to build for the Raspberry PI 4. 

I finally ended up adding: 

set(PLATFORM DRM)
set(GRAPHICS API_OPENGL_ES2)

To the bottom of the CMakeOptions.txt file. 

I always end up with the atomic library missing when it tries to build the examples. I edited the link.txt files in all of the directories below the raylib/build/examples/CMakeFiles directory adding -latomic to the end of build lines.

The examples now build, but when I run them I get an error saying:

INFO: DISPLAY: No graphic card set, trying platform-gpu-card
WARNING: DISPLAY: Failed to find a suitable DRM connector mode
FATAL: Failed to initialize Graphic Device

Any idea how to get this to work? 

Thanks,
/Chris

**Comment by archie2x** [comment]
1) Pull latest. There was a bug with cmake -DPLATFORM=DRM. I don't know if that was your problem but it works for me now.

2) Try setting the resolution of your display in the example. For me, I have a 480x320 display so I have to modify e.g. shapes_logo_raylib.c

```
diff --git a/examples/shapes/shapes_logo_raylib.c b/examples/shapes/shapes_logo_raylib.c
index 48005c7a..3521934d 100644
--- a/examples/shapes/shapes_logo_raylib.c
+++ b/examples/shapes/shapes_logo_raylib.c
@@ -20,8 +20,8 @@ int main(void)
 {
     // Initialization
     //--------------------------------------------------------------------------------------
-    const int screenWidth = 800;
-    const int screenHeight = 450;
+    const int screenWidth = 480;
+    const int screenHeight = 320;
 
     InitWindow(screenWidth, screenHeight, "raylib [shapes] example - raylib logo using shapes");
 
@@ -58,4 +58,4 @@ int main(void)
     //--------------------------------------------------------------------------------------
 
     return 0;
-}
\ No newline at end of file
```

3) If that doesn't work... more info please?

This is Raspbian ? What's the display?
